File: platform/src/main/java/com/flickmatch/platform/graphql/builder/MessageBuilder.java
package com.flickmatch.platform.graphql.builder;

import com.flickmatch.platform.proxy.TwilioProxy;
import lombok.extern.log4j.Log4j2;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.twilio.rest.api.v2010.account.Message;
import com.twilio.type.PhoneNumber;

@Log4j2
@Service
public class MessageBuilder {

    @Autowired
    private TwilioProxy twilioProxy;

    public void sendMessage(String from, String to, String message) {
        log.info("Sending message from: {} to: {}", from, to);
        try {
            twilioProxy.initTwilio();
//            System.out.println(message);
            Message whatsappMessage = Message.creator(
                    new PhoneNumber("whatsapp:" + to.trim()),   // Recipient
                    new PhoneNumber("whatsapp:" + from.trim()), // Twilio WhatsApp Business Number
                    message
            ).create();

            log.info("WhatsApp message sent successfully, SID: {}", whatsappMessage.getSid());

        } catch (Exception e) {
            log.error("Failed to send message: {}", e.getMessage(), e);
        }

    }
}

File: python/twilio/message_builder.py
from twilio.base.exceptions import TwilioRestException
from twilio.rest.api.v2010.account.message import MessageInstance
import logging
import json

logger = logging.getLogger(__name__)


class MessageBuilder:
    def __init__(self, twilio_proxy):
        self.twilio_proxy = twilio_proxy

    def send_message(self, from_number: str, to_number: str, sid: str, first_name: str) -> str:
        sanitized_from_number = from_number.replace('\n', '').replace('\r', '')
        sanitized_to_number = to_number.replace('\n', '').replace('\r', '')
        sanitized_sid = sid.replace('\n', '').replace('\r', '')
        sanitized_first_name = first_name.replace('\n', '').replace('\r', '')
        logger.info(
            f"Sending message from: {sanitized_from_number} to: {sanitized_to_number}")
        try:
            client = self.twilio_proxy.init_twilio()

            message_response: MessageInstance = client.messages.create(
                content_sid=sanitized_sid,
                from_=f"whatsapp:{sanitized_from_number}",
                to=f"whatsapp:{sanitized_to_number}",
                content_variables=json.dumps({"first_name": first_name}),
            )

            logger.info(
                f"WhatsApp message sent successfully, SID: {message_response.sid}")
            return message_response.sid
        except TwilioRestException as e:
            logger.error(f"Failed to send message: {str(e)}")
            raise e
        except Exception as e:
            logger.error(f"Unexpected error: {str(e)}")
            raise e
